infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationSinkClientRouter.java

```diff
@@ -0,0 +1,555 @@
+     */
+    public void connect() {
+        LogReplicationPluginConfig config = new LogReplicationPluginConfig(this.pluginFilePath);
+        File jar = new File(config.getTransportAdapterJARPath());
+
+        try (URLClassLoader child = new URLClassLoader(new URL[]{jar.toURI().toURL()}, this.getClass().getClassLoader())) {
+            // Instantiate Channel Adapter (external implementation of the channel / transport)
+            Class adapterType = Class.forName(config.getTransportClientClassCanonicalName(), true, child);
+            channelAdapter = (IClientChannelAdapter) adapterType.getDeclaredConstructor(String.class, ClusterDescriptor.class, LogReplicationSourceClientRouter.class, LogReplicationSinkClientRouter.class)
+                    .newInstance(this.localClusterId, remoteClusterDescriptor, null, this);
+            log.info("Connect asynchronously to remote cluster {} and session {} ", remoteClusterDescriptor.getClusterId(), this.session);
+
+            // When connection is established to the remote leader node, the remoteLeaderConnectionFuture will be completed.
+            channelAdapter.connectAsync(session);
+        } catch (Exception e) {
+            log.error("Fatal error: Failed to initialize transport adapter {}", config.getTransportClientClassCanonicalName(), e);
+            throw new UnrecoverableCorfuError(e);
+        }
+    }
+
+    /**
+     * Connection Up Callback.
+     *
+     * @param nodeId id of the remote node to which connection was established.
+     */
+    @Override
+    public synchronized void onConnectionUp(String nodeId) {
+        log.info("Connection established to remote node {}.", nodeId);
+        this.verifyLeadership.input(new LogReplicationRuntimeEvent(LogReplicationRuntimeEvent.LogReplicationRuntimeEventType.ON_CONNECTION_UP, nodeId));
+    }
+
+    /**
+     * Connection Down Callback.
+     *
+     * @param nodeId id of the remote node to which connection came down.
+     */
+    @Override
+    public synchronized void onConnectionDown(String nodeId) {
+        log.info("Connection lost to remote node {} on cluster {}", nodeId, this.session.getSourceClusterId());
+        this.verifyLeadership.input(new LogReplicationRuntimeEvent(LogReplicationRuntimeEvent.LogReplicationRuntimeEventType.ON_CONNECTION_DOWN, nodeId));
+        // Attempt to reconnect to this endpoint
+        channelAdapter.connectAsync(nodeId, session);
+    }
+
+    /**
+     * Channel Adapter On Error Callback
+     */
+    public synchronized void onError(Throwable t) {
+        // no op
+        log.error("Unrecoverable state due to error {}", t.getMessage());
+    }
+
+    /**
+     * Cluster Change Callback.
+     *
+     * @param clusterDescriptor remote cluster descriptor
+     */
+
+    public synchronized void onClusterChange(ClusterDescriptor clusterDescriptor) {
+        //no-op. Any topology change notification goes to discovery-service
```

Comment:
Actually no..I had a todo for myself in the LogRepicationBaseSourceRouter for the same method.
Its not needed for GRPC, since the clusterChange here refers to a node being added or deleted or similar..ie., something within a cluster changed.

To be generic, I am now passing the call to the IClientAdapter. We can process/ignore it as required there.